watertap/unit_models/boron_removal.py

```diff
@@ -766,13 +772,23 @@ def initialize_build(
         # Solve unit
         with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
             res = opt.solve(blk, tee=slc.tee)
+            # occasionally it might be worth retrying a solve
+            if not check_optimal_termination(res):
```

Comment:
I am not sure if this is a good idea or not - for one if you are seeing this sort of behaviour it tends to indicate a model issue that really should be fixed.